### PR TITLE
[MIG] website_apps_store: Migration to 12.0

### PR DESCRIPTION
**Pull request by martintorresvm**
cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@victormmtorres it seems there's something strange in your commit email address. Please check for CLA, and both runbot and Travis are red.

File: website_apps_store/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        for product in self:
+            product.total_download_count = sum(
+                product.mapped('product_variant_ids.download_count'))
+
+    total_download_count = fields.Integer(
+        compute="_compute_total_download_count", store=True)
+
+    def get_author_details(self):
+        author_ids = []
+        for variant in self.product_variant_ids:
```
Comment (umiphos) [inline website_apps_store/models/product_template.py]: can you try:

```suggestion
        author_ids = variant in self.mapped('product_variant_ids.app_author_ids').ids
```

Comment (pedrobaeza) [reply website_apps_store/models/product_template.py]: Not exactly... it should be directly:
```python
return self.mapped('product_variant_ids.app_author_ids')
```

**Comment by bizzappdev** [comment]
@victormmtorres one fix has been added in v11.
please also include those fix in v12
https://github.com/OCA/apps-store/pull/64

**Comment by pedrobaeza** [comment]
@victormmtorres check also runbot and Travis status

**Comment by martintorresvm** [comment]
I've include #64 and missed dependencies hope Travis and Runbot pass

**Comment by martintorresvm** [comment]
Changes done @pedrobaeza

**Review by MiquelRForgeFlow — APPROVED** [review]
code review :+1: 

(although I don't understand the changes in `test_website_apps_store.py`)

File: website_apps_store/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                     'ilike', srch),
+                    ('product_variant_ids.app_author_ids.name', 'ilike', srch),
+                    ('product_variant_ids.app_summary', 'ilike', srch)]
+        if category:
+            domain += [('public_categ_ids', 'child_of', int(category))]
+
+        if attrib_values:
+            attrib = None
+            ids = []
+            for value in attrib_values:
```
Comment (Tardo) [inline website_apps_store/controllers/main.py:45]: This loop, for some reason, looks strange... but i don't tested it.

File: website_apps_store/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                if not attrib:
+                    attrib = value[0]
+                    ids.append(value[1])
+                elif value[0] == attrib:
+                    ids.append(value[1])
+                else:
```
Comment (Tardo) [inline website_apps_store/controllers/main.py:51]: Why need this? This do the same that "if not attrib" but split the domain in more 'criterias' no?

All conditions do the same... add to ids the "value[1]".

File: website_apps_store/controllers/main.py (hunk 2 above)
Comment (martintorresvm) [reply website_apps_store/controllers/main.py:45]: This PR scope is migration so the logic should not change without a reason (normally changes on Odoo base code, new API, changes in guidelines,...etc) but feel free to suggest improvements with comment code.

Basicly the module should work as the previous version

File: website_apps_store/controllers/main.py (hunk 3 above)
Comment (martintorresvm) [reply website_apps_store/controllers/main.py:51]: As I said this PR scope is migration

**Review by pedrobaeza — APPROVED** [review]
Let's move on as this module is inherited from others, but should work.

**Pull request merged**